prow/build_test_update.md

```diff
@@ -59,22 +59,84 @@ kubectl apply -f prow/cluster/hook_deployment.yaml
+
+ProwJobs can also be manually triggered by generating a YAML ProwJob CRD resource
+with mkpj and deploying that directly to the Prow cluster, however this pattern
+is generally not recommended. It requires the use of direct prod cluster access,
+allows ProwJobs to run in prod without passing presubmit validation, and can result
+in malformed ProwJobs CRDs that can jam some of Prow's core service components.
+See [How to manually run a given job on Prow](#how-to-manually-run-a-given-job-on-prow)
+below if you need to do this.
 
 ### Running a ProwJob Locally
 
-[Phaino](/prow/cmd/phaino) lets you interactively mock and run the job locally on your workstation
-in a docker container. Detailed instructions can be found in Phaino's [Readme](/prow/cmd/phaino/README.md)
+#### Using pj-on-kind.sh
+[pj-on-kind.sh] is a bash script that runs ProwJobs locally as pods in a [Kind] cluster.
+The script does the following:
+1. Installs [mkpj], [mkpod], and [Kind] if they are not found in the path. A [Kind]
+cluster named `mkpod` is created if one does not already exist.
+1. Uses [mkpj] to generate a YAML ProwJob CRD given job name, config, and git refs (if applicable).
+1. Uses [mkpod] to generate a YAML Pod resource from the ProwJob CRD. This Pod will
+be decorated with the pod utilities if needed and will exactly match what would be
+applied in prod with two exceptions:
+	1. The job logs, metadata, and artifacts will be copied to disk rather than
+	uploaded to GCS. By default these files are copied to `/etc/prowjob-out/<job-name>/<build-id>/`
+	on the host machine.
+	1. Any volume mounts may be substituted for `emptyDir` or `hostPath` volumes at the
+	interactive prompt to replace dependencies that are only available in prod.
+	Note: In order for `hostPath` to reach the host and not just the Kind "node" container,
+	use paths under `/etc/kind-node` or set `$NODE_DIR` before the mkpod cluster is created.
+1. Applies the Pod to the [Kind] cluster and starts watching it (interrupt whenever,
+this is for convenience). At this point the Pod will start running if configured
+correctly.
+
+Once the Pod has been applied to the cluster you can wait for it to complete and output
+results to the output directory, or you can interact with it using kubectl by first
+running `export KUBECONFIG="$(kind get kubeconfig-path --name=mkpod)"`.
+
+Requirements: [Go], [Docker], and [kubectl] must be installed before using this script.
+The ProwJob must use `agent: kubernetes` (the default, runs ProwJobs as Pods).
+
+##### pj-on-kind.sh for specific Prow instances
+Each Prow instance can supply a preconfigured variant of pj-on-kind.sh that properly
+defaults the config file locations. [Example](https://github.com/istio/test-infra/blob/01167b0dc9cb19bee40aa8dff958f526cfeeb570/prow/pj-on-kind.sh)
+for [prow.istio.io](https://prow.istio.io).
+To test ProwJobs for the [prow.k8s.io] instance use [`config/pj-on-kind.sh`](/config/pj-on-kind.sh).
+
+##### Example
```

Comment:
```suggestion
##### Example
This command runs the Prow job [`pull-test-infra-yamllint`](https://github.com/kubernetes/test-infra/blob/170921984a34ca40f2763f9e71d6ce6e033dec03/config/jobs/kubernetes/test-infra/test-infra-presubmits.yaml#L94-L107) locally
```

This explicitly calls out that the command is a Prow Job name, and that it's identical to the name of the configuration objects in `config/jobs/...`